oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -200,6 +211,36 @@ public void createAssertion_correct() throws IOException {
     assertEquals(Joiner.on(' ').join(scopes), payload.get("scope"));
   }
 
+  @Test
+  public void createAssertionWithExplicitProvider_correct() throws IOException {
+    PrivateKey privateKey = ServiceAccountCredentials.privateKeyFromPkcs8(PRIVATE_KEY_PKCS8);
+    List<String> scopes = Arrays.asList("scope1", "scope2");
+    ServiceAccountCredentials credentials =
+            ServiceAccountCredentials.newBuilder()
+                    .setClientId(CLIENT_ID)
+                    .setClientEmail(CLIENT_EMAIL)
+                    .setPrivateKey(privateKey)
+                    .setPrivateKeyId(PRIVATE_KEY_ID)
+                    .setScopes(scopes)
+                    .setServiceAccountUser(USER)
+                    .setProjectId(PROJECT_ID)
+                    .setSignatureProvider(defaultRsaSignatureProvider)
+                    .build();
+
+    JsonFactory jsonFactory = OAuth2Utils.JSON_FACTORY;
+    long currentTimeMillis = Clock.SYSTEM.currentTimeMillis();
+    String assertion = credentials.createAssertion(jsonFactory, currentTimeMillis, null);
+
+    JsonWebSignature signature = JsonWebSignature.parse(jsonFactory, assertion);
+    JsonWebToken.Payload payload = signature.getPayload();
+    assertEquals(CLIENT_EMAIL, payload.getIssuer());
+    assertEquals(OAuth2Utils.TOKEN_SERVER_URI.toString(), payload.getAudience());
+    assertEquals(currentTimeMillis / 1000, (long) payload.getIssuedAtTimeSeconds());
```

Comment:
This looks like it might be flaky if the system is loaded since it depends on the system clock. Try com.google.api.client.testing.http.FixedClock  instead.

Reply:
I would suggest that being changed in a separate change, as it is the pattern throughout that file.

Reply:
You don't have to change the existing methods in this PR--in fact, please don't since it would make it harder to review--but we do try to not copy old bad habits into new code.

Reply:
Roger that, will fix.